gma/src/include/firebase/gma/ump/types.h

```diff
@@ -0,0 +1,209 @@
+  bool tag_for_under_age_of_consent_;
+  bool has_debug_settings_;
+  bool has_tag_for_under_age_of_consent_;
+};
+
+/// This is a platform specific datatype that is required to show a consent form
+/// on screen.
+///
+/// The following defines the datatype on each platform:
+/// <ul>
+///   <li>Android: A `jobject` which references an Android Activity.</li>
+///   <li>iOS: An `id` which references an iOS UIViewController.</li>
+/// </ul>
+#if FIREBASE_PLATFORM_ANDROID
+/// An Android Activity from Java.
+typedef jobject FormParent;
+#elif FIREBASE_PLATFORM_IOS || FIREBASE_PLATFORM_TVOS
+/// A pointer to an iOS UIViewController.
+typedef id FormParent;
+#else
+/// A void pointer for stub classes.
+typedef void* FormParent;
+#endif  // FIREBASE_PLATFORM_ANDROID, FIREBASE_PLATFORM_IOS,
+        // FIREBASE_PLATFORM_TVOS
+
+/// Consent status values.
+enum ConsentStatus {
+  /// Unknown status, e.g. prior to calling Request, or if the request fails.
+  kConsentStatusUnknown = 0,
+  /// Consent is required, but not obtained
+  kConsentStatusRequired,
+  /// Consent is not required
+  kConsentStatusNotRequired,
+  /// Consent was required, and has been obtained
+  kConsentStatusObtained
+};
+
+/// Errors that can occur during a RequestConsentInfoUpdate operation.
+enum ConsentRequestError {
+  /// The operation succeeded.
+  kConsentRequestSuccess = 0,
+  /// Invalid GMA App ID specified in AndroidManifest.xml or Info.plist.
+  kConsentRequestErrorInvalidAppId,
+  /// A network error occurred.
+  kConsentRequestErrorNetwork,
+  /// The tag for age of consent was not set. You must call
+  /// `ConsentRequestParameters::set_tag_for_under_age_of_consent()` before the
+  /// request.
+  kConsentRequestErrorTagForAgeOfConsentNotSet,
+  /// An internal error occurred.
+  kConsentRequestErrorInternal,
+  /// This error is undocumented.
+  kConsentRequestErrorCodeMisconfiguration,
+  /// An unknown error occurred.
+  kConsentRequestErrorUnknown,
+  /// The operation is already in progress. Use
+  /// `ConsentInfo::RequestConsentInfoUpdateLastResult()`
+  /// to get the status.
+  kConsentRequestErrorOperationInProgress
+};
+
+/// Status of the consent form, whether it is available to show or not.
+enum ConsentFormStatus {
+  /// Status is unknown. Call `ConsentInfo::RequestConsentInfoUpdate()` to
+  /// update
+  /// this.
+  kConsentFormStatusUnknown = 0,
+  /// The consent form is unavailable. Call `ConsentInfo::LoadConsentForm()` to
+  /// load it.
+  kConsentFormStatusUnavailable,
+  /// The consent form is available. Call `ConsentInfo::ShowConsentForm()` to
+  /// display it.
+  kConsentFormStatusAvailable,
+};
+
+/// Errors when loading or showing the consent form.
+enum ConsentFormError {
+  /// The operation succeeded.
+  kConsentFormSuccess = 0,
+  /// The load request timed out. Try again.
+  kConsentFormErrorTimeout,
+  /// Failed to show the consent form because it has not been loaded.
+  kConsentFormErrorNotLoaded,
```

Comment:
The iOS error enum has a value UMPFormErrorCodeUnavailable, which indicates that the form is not available (possibly because it has not been loaded)
We might want to rename this to kConsentFormErrorUnavailable and use it for any case where there is no form to show.

Reply:
(In the API proposal I removed NotLoaded and added Unavailable)

Reply:
I removed NotLoaded and added Unavailable.